opa/pkg/main.go

```diff
@@ -0,0 +1,324 @@
+							Category:    "Reliability",
+						})
+						logrus.Warnf("Error while parsing Rego %+v", err)
+						return actionItems, nil
+					}
+					obj.Object["parameters"] = checkInstance.Spec.Parameters
+					// TODO Find another way to get parameters in - Should they be a function or input?
+					// TODO Caching
+					evaluatedInput := rego.EvalInput(obj.Object)
+					results, err := query.Eval(ctx, evaluatedInput)
+					if err != nil {
+						return nil, err
+					}
+					newItems, err := processResults(obj, results, check, checkInstance)
+					if err != nil {
+						actionItems = append(actionItems, ActionItem{
+							Title:       checkInstance.Name,
+							Description: "An error occured processing the results of this check.",
+							Remediation: "Make sure that the return values are all correct.",
+							Severity:    0.4,
+							Category:    "Reliability",
+						})
+						logrus.Warnf("Error while parsing results %+v", err)
+						return actionItems, nil
+					}
+					actionItems = append(actionItems, newItems...)
+				}
+			}
+		}
+	}
+	return actionItems, nil
+}
+
+func getOutputArray(results rego.ResultSet) []interface{} {
+	returnSet := make([]interface{}, 0)
+
+	for _, result := range results {
+		for _, pack := range result.Bindings["results"].(map[string]interface{}) {
+			for _, outputArray := range pack.(map[string]interface{}) {
+				for _, output := range outputArray.([]interface{}) {
+					returnSet = append(returnSet, output)
+				}
+			}
+		}
+	}
+	return returnSet
+}
+
+func processResults(resource unstructured.Unstructured, results rego.ResultSet, check customCheck, checkInstance customCheckInstance) ([]ActionItem, error) {
+	actionItems := make([]ActionItem, 0)
+	instanceOutput := checkInstance.Spec.Output
+	checkOutput := check.Spec.Output
+	for _, output := range getOutputArray(results) {
+		severity := checkOutput.Severity
+		title := checkOutput.Title
+		remediation := checkOutput.Remediation
+		category := checkOutput.Category
+		if instanceOutput.Severity != nil {
+			severity = instanceOutput.Severity
+		}
+		if instanceOutput.Title != nil {
+			title = instanceOutput.Title
+		}
+		if instanceOutput.Remediation != nil {
+			remediation = instanceOutput.Remediation
+		}
+		if instanceOutput.Category != nil {
+			category = instanceOutput.Category
+		}
+		strMethod, ok := output.(string)
+		var description string
+		if ok {
+			description = strMethod
+		} else {
+			mapMethod, ok := output.(map[string]interface{})
+			if ok {
+				description = mapMethod["description"].(string)
+				if mapMethod["severity"] != nil {
+					severityFloat, err := strconv.ParseFloat(mapMethod["severity"].(string), 64)
+					if err != nil {
+						return nil, err
+					}
+					severity = &severityFloat
+				}
+				if mapMethod["title"] != nil {
+					titleString := mapMethod["title"].(string)
+					title = &titleString
+				}
+
+				if mapMethod["remediation"] != nil {
+					remediationString := mapMethod["remediation"].(string)
```

Comment:
should we worry about these not being strings, e.g. if the user did something wrong? Go will `panic` if so